mutation.hh

```diff
@@ -276,6 +267,10 @@ std::optional<stop_iteration> consume_clustering_fragments(schema_ptr s, mutatio
         }
         bool emit_rt = rts_it != rts_end;
         if (rts_it != rts_end) {
+            if (reverse == consume_in_reverse::yes && !cookie.reversed_range_tombstone) {
+                rts_it->tombstone().reverse();
```

Comment:
This will modify `partition` in-place, which may be unexpected from "consume". You can try to consume mutation multiple times.

Maybe instead of `reversed_range_tombstone = true`, cache the resulting tombstone.

Reply:
My first iteration used an optional reversed range tombstone in the cookie, but Botond objected to that since it complicated the code.

Since we're already consuming the partition in place, moving the range tombstone and rows away from it.
See https://github.com/scylladb/scylladb/blob/60e8f5743cc777882c6b53fa04a0e82c8ae862b2/mutation.hh#L287-L295
I don't see a problem with reversing the rt in place.

Reply:
@tgrabiec `consume()` is destructive and you can only do it once. Once you call `consume()` on a mutation, the only thing you can do with it is to finish consuming and/or destroy it. This is why it has a `&&` ref spec.

Reply:
Right, it's fine then.